uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/instances/leave-request.instance.js

```diff
@@ -11,6 +11,38 @@ define([
   instances.factory('LeaveRequestInstance', ['$q', 'checkPermissions', 'OptionGroup',
     'shared-settings', 'ModelInstance', 'LeaveRequestAPI',
     function ($q, checkPermissions, OptionGroup, sharedSettings, ModelInstance, LeaveRequestAPI) {
+      /**
+       * Amends the first and last days of the balance by setting values from the
+       * selected time deductions. It also re-calculates the total amount.
+       *
+       * @param  {LeaveRequestInstance} request
+       * @param  {Object} balanceChange
+       * @return {Promise} resolves with amended balance breakdown
+       */
+      function amendBalanceChangeBreakdownForHoursUnit (request, balanceChange) {
+        var firstDay, lastDay;
+
+        balanceChange = _.cloneDeep(balanceChange);
```

Comment:
Why is cloning necessary?

Reply:
We are passing an object and I don't want to mutate it.

Reply:
Please see above